### PR TITLE
solved

### PR DESCRIPTION
**Pull request by VladSkoromnyi**

### CODE REVIEW

**Review by vlasiuk-anatolii — APPROVED** [review]
Good job!

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const firstState = { ...state };
+  const changeState = [];
+
+  function assigned(index, to, from) {
+    changeState[index] = Object.assign(to, from);
+  }
```
Comment (Anna-Tkachenko) [inline src/transformStateWithClones.js:15]: This function is redundant, you can use destructuring for creating object copy and add it to array
array.push(...object)

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,33 @@
...
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const key in actions[i].extraData) {
+        firstState[key] = actions[i].extraData[key];
+      }
+      assigned(i, {}, firstState);
```
Comment (Anna-Tkachenko) [inline src/transformStateWithClones.js:22]: Don't repeat yourself: you use this functionality 3 times in your code, move it at the end of cycle

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
Use switch / case if you need to do some logic depending on the value of some variable

**Review by Anna-Tkachenko — APPROVED** [review]
Good job! 👍

**Pull request closed without merging**